components/discovery/KidFilter.jsx

```diff
@@ -0,0 +1,92 @@
+"use client";
+
+import React, { useState, useEffect } from "react";
+import HobbySelect from "../general/HobbySelect";
+
+export default function KidFilter({
+  setHobbies,
+  hobbies,
+  setAge,
+  age,
+  setPronouns,
+  pronouns,
+  filter,
+}) {
+  const [hobbyFilter, setHobbiesFilter] = useState(hobbies || []);
+  const [ageFilter, setAgeFilter] = useState(age || 0);
+  const [pronounsFilter, setPronounsFilter] = useState(pronouns || "");
+
+  useEffect(() => {
+    setHobbiesFilter(hobbies || []); 
+    setAgeFilter(age || 0); 
+    setPronounsFilter(pronouns || ""); 
+  }, [age, pronouns, hobbies]);
+
+  const applyFilter = (e) => {
+    e.preventDefault();
+    filter(ageFilter, hobbyFilter, pronounsFilter);
+  };
+
+  const clearFilter = () => {
+    setHobbies(null);
+    setAge(null);
+    setPronouns(null);
+  };
+
+  return (
+    <div className=" bg-white flex flex-col my-14">
+      <form className="flex flex-col gap-6">
+        <HobbySelect setHobbies={setHobbiesFilter} hobbies={hobbyFilter} />
+        <div>
+          <label
+            htmlFor="pronouns"
+            className="text-sm font-medium text-gray-700 block mb-2"
+          >
+            Pronouns
+          </label>
+          <select
+            id="pronouns"
+            value={pronounsFilter}
+            onChange={(e) => setPronounsFilter(e.target.value)}
+            className="w-full p-3 border border-gray-300 rounded-md text-black"
+          >
+            <option value="">Select your pronouns</option>
```

Comment:
It's the child's attributes, not the user's:
```suggestion
            <option value="">Child's gender</option>
```